### PR TITLE
#831: [Disk Manager] validate crc32 for each block; log new crc32 for each block after writing in tests;

### PR DESCRIPTION
**Pull request by BarkovBG**
https://github.com/ydb-platform/nbs/issues/831

### CODE REVIEW

**Comment by BarkovBG** [comment]
https://github.com/ydb-platform/nbs/issues/831

File: cloud/disk_manager/internal/pkg/clients/nbs/testing_client.go [hunk 1]
```diff
@@ -149,33 +159,71 @@ func (c *client) CalculateCrc32WithEncryption(
...
+	)
 }
 
 func (c *client) ValidateCrc32WithEncryption(
+	ctx context.Context,
 	diskID string,
 	contentSize uint64,
 	encryption *types.EncryptionDesc,
 	expectedCrc32 uint32,
+	expectedBlocksCrc32 []uint32,
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/clients/nbs/testing_client.go:189]: expectedBlockCrc32s

File: cloud/disk_manager/internal/pkg/facade/disk_service_test/disk_service_test.go [hunk 2]
```diff
@@ -131,9 +131,11 @@ func TestDiskServiceCreateDiskFromImageWithForceNotLayered(t *testing.T) {
 
 	nbsClient := testcommon.NewNbsClient(t, ctx, "zone-a")
 	err = nbsClient.ValidateCrc32(
+		ctx,
 		diskID,
 		testcommon.GetRawImageSize(t),
 		testcommon.GetRawImageCrc32(t),
+		[]uint32{}, // We do not known expectedBlocksCrc32 for image.
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/facade/disk_service_test/disk_service_test.go:138]: We do not know ...

File: cloud/disk_manager/internal/pkg/facade/testcommon/common.go [hunk 3]
```diff
@@ -407,22 +408,44 @@ func FillEncryptedDisk(
...
+			}
+
+			logging.Debug(
+				ctx,
+				"%v block with index %v crc32 now is %v",
+				diskID,
+				startIndex+uint64(blockIndex),
+				blockAcc.Sum32(),
+			)
+			blocksCrc32 = append(blocksCrc32, blockAcc.Sum32())
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/facade/testcommon/common.go:444]: blockCrc32s

File: cloud/disk_manager/internal/pkg/clients/nbs/testing_client.go [hunk 4]
```diff
@@ -149,33 +159,71 @@ func (c *client) CalculateCrc32WithEncryption(
...
 ) error {
 
-	actualCrc32, err := c.CalculateCrc32WithEncryption(diskID, contentSize, encryption)
+	actualCrc32, actualBlocksCrc32, err := c.CalculateCrc32WithEncryption(
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/clients/nbs/testing_client.go:192]: actualBlockCrc32s

File: cloud/disk_manager/internal/pkg/facade/testcommon/common.go [hunk 5]
```diff
@@ -545,7 +582,7 @@ func CreateImage(
 	require.Equal(t, int64(imageSize), response.Size)
 	require.Equal(t, int64(storageSize), response.StorageSize)
 
-	return crc32, storageSize
+	return crc32, blocksCrc32, storageSize
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/facade/testcommon/common.go:585]: blockCrc32s

and others

File: cloud/disk_manager/internal/pkg/facade/disk_service_test/disk_service_test.go [hunk 6]
```diff
@@ -131,9 +132,13 @@ func TestDiskServiceCreateDiskFromImageWithForceNotLayered(t *testing.T) {
...
 	nbsClient := testcommon.NewNbsClient(t, ctx, "zone-a")
 	err = nbsClient.ValidateCrc32(
+		ctx,
 		diskID,
-		testcommon.GetRawImageSize(t),
-		testcommon.GetRawImageCrc32(t),
+		nbs.DiskContentInfo{
+			ContentSize: testcommon.GetRawImageSize(t),
+			Crc32:       testcommon.GetRawImageCrc32(t),
+			BlockCrc32s: []uint32{}, // We do not known expectedBlocksCrc32 for image.
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/facade/disk_service_test/disk_service_test.go:140]: known -> know

**Review by SvartMetal — APPROVED** [review]

**Review by drbasic — APPROVED** [review]

**Pull request merged**